<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <param name="use_sim_time" value="false" />

    <arg name="wait_det_flag" default="true" />
    <arg name="camera" default="/camera/color" />

    <arg name="cloud_topic" default="/os_cloud_node/points" />
    <arg name="odom_topic" default="/BLIO/odom_gicp_global_prediction" />
    <arg name="info_topic" default="$(arg camera)/camera_info" />
    <arg name="camera_topic" default="$(arg camera)/image_raw/compressed" />
    <arg name="compressed" default="true"/>
    <arg name="camera_frame" default="camera" />
    <arg name="cloud_frame" default="lidar" />
    <arg name="cloud_out_topic" default="$(arg cloud_topic)/rgb" />
    <arg name="image_out_topic" default="$(arg camera)/projected" />
    
    
    <!-- DRONE_IDX 0 FOR ADD, 1 FOR DRONEBOT -->
    <arg name="drone_idx" default="1" />

    <!-- <node
        pkg="tf"
        type="static_transform_publisher"
        name="cloud_camera_tf"
        output="screen"
        args="-0.00706296 -0.11697807 -0.24420192 0.52218648 -1.52973857  0.97920602 $(arg camera_frame) $(arg cloud_frame) 50" />

    <node
        pkg="tf"
        type="static_transform_publisher"
        name="tf_baselink2ossensor"
        output="screen"
        args="0 0 0 0 0 0 base_link os_sensor 10" /> -->

    <node pkg="pycoral_ros" type="add_detection" name="add_detection" output="screen">
        <param name="wait_det_flag" value="$(arg wait_det_flag)" />
        <param name="cloud_topic" value="$(arg cloud_topic)"/>
        <param name="camera_topic" value="$(arg camera_topic)"/>
        <param name="info_topic" value="$(arg info_topic)"/>
        <param name="cloud_out_topic" value="$(arg cloud_out_topic)"/>
        <param name="image_out_topic" value="$(arg image_out_topic)"/>
        <param name="compressed" value="$(arg compressed)"/>
        <param name="camera_frame" value="$(arg camera_frame)"/>
        <param name="cloud_frame" value="$(arg cloud_frame)"/>
        <param name="drone_idx" value="$(arg drone_idx)"/>
        <remap from="/odom" to="$(arg odom_topic)"/>

        <!-- max_objects -> maximum object number per category -->
        <!-- cluster_tolerance -> euclidian cluster tolerance distance per category -->
        <!-- 
            Current categories
            0. Person
                # 10 
            1. Stretcher
                # 2
            2. Computer
                # 3
            3. Desk
                # 4
            4. Communication Equipment
                # 1
            5. Weapon storage
                # 1
            6. Barbed wire
                # 1
         -->
        <param name="num_categories" value="7"/>

        <!-- 0. PERSON -->
        <param name="max_objects_0" value="25"/>
        <param name="min_objects_0" value="8"/> 
        <param name="min_points_0" value="5"/>
        <param name="cluster_tolerance_0" value="0.3"/>
        <param name="min_intensity_0" value="0.5"/>

        <!-- 1. PERSON ROTATED -->
        <param name="max_objects_1" value="6"/>
        <param name="min_objects_1" value="-1"/>
        <param name="min_points_1" value="3"/> 
        <param name="cluster_tolerance_1" value="3"/>
        <param name="min_intensity_1" value="0.45"/>
        
        <!-- 2. STRETCHER -->
        <param name="max_objects_2" value="6"/>
        <param name="min_objects_2" value="-1"/>
        <param name="min_points_2" value="5"/> 
        <param name="cluster_tolerance_2" value="3"/>
        <param name="min_intensity_2" value="0.5"/>

        <!-- 3. COMPUTER -->
        <param name="max_objects_3" value="6"/>
        <param name="min_objects_3" value="2"/>
        <param name="min_points_3" value="5"/> 
        <param name="cluster_tolerance_3" value="1.4"/>
        <param name="min_intensity_3" value="0.5"/>

        <!-- 4. COMMUNICATION EQUIPMENT -->
        <param name="max_objects_4" value="6"/>
        <param name="min_objects_4" value="-1"/>
        <param name="min_points_4" value="4"/> 
        <param name="cluster_tolerance_4" value="1"/>
        <param name="min_intensity_4" value="0.5"/>

        <!-- 5. WEAPON STORAGE -->
        <param name="max_objects_5" value="4"/>
        <param name="min_objects_5" value="-1"/>
        <param name="min_points_5" value="4"/> 
        <param name="cluster_tolerance_5" value="4"/>
        <param name="min_intensity_5" value="0.5"/>

        <!-- 6. BARBER WIRE -->
        <param name="max_objects_6" value="4"/>
        <param name="min_objects_6" value="-1"/>
        <param name="min_points_6" value="4"/> 
        <param name="cluster_tolerance_6" value="4"/>
        <param name="min_intensity_6" value="0.5"/>

    </node>

    <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pycoral_ros)/rviz/detection.rviz" /> -->

    <include file="$(find pycoral_ros)/launch/detect.launch"/>
</launch>